gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/MaxLocalStep.java

```diff
@@ -34,23 +33,24 @@
  * @author Marko A. Rodriguez (http://markorodriguez.com)
  * @author Daniel Kuppitz (http://gremlin.guru)
  */
-public final class MaxLocalStep<E extends Comparable, S extends Iterable<E>> extends ScalarMapStep<S, E> {
+public final class MaxLocalStep<E extends Comparable, S extends Iterable<E>> extends FlatMapStep<S, E> {
```

Comment:
This would extend just abstract MapStep class too right? We wont ever emit more than single solution out which is usually when FlatMapStep is used for? (Same for other local step in the PR).

Reply:
Introduced `OptionalScalarMap` to support no output case

Reply:
I reverted `OptionalScalarMap`, in some case TinkerPop still wants to emit `null` so we can't always filter the input.